src/app/clusters/icd-management-server/icd-management-server.cpp

```diff
@@ -169,6 +169,7 @@ class IcdManagementFabricDelegate : public chip::FabricTable::Delegate
         ICDMonitoringTable table(Server::GetInstance().GetPersistentStorage(), fabricIndex, supported_clients,
                                  Server::GetInstance().GetSessionKeystore());
         table.RemoveAll();
+        ICDManagementServer::GetInstance().TableIsEmptyForFabric();
```

Comment:
It is unfortunate that this is going via a global singleton instance, rather than via something injectable via Server or such.

Reply:
Created an issue to track this when we get to the clean up
https://github.com/project-chip/connectedhomeip/issues/30240